solarforecastarbiter/metrics/deterministic.py

```diff
@@ -188,6 +173,8 @@ def pearson_correlation_coeff(obs, fx):
         The correlation coefficient (r [-]) of the observations and forecasts.
 
     """
+    if len(obs) == 1 or len(fx) == 1:
```

Comment:
should this be `and` instead of `or`?

Reply:
To elaborate... I think `and` is right because we want to fail in some way if there's a length mismatch. That failure will happen in `pearsonr`:

```python
    n = len(x)
    if n != len(y):
        raise ValueError('x and y must have the same length.')

    if n < 2:
        raise ValueError('x and y must have length at least 2.')
```

Reply:
Hmm, I think this should be revamped. ``sp.stats.pearsonr`` raises a ValueError when ``len(fx) != len(obs)`` or ``len(fx) < 2``, so probably want to test those conditions.

Reply:
`pearsonr` also raises a `ValueError` if there are any nans in the data. Debugging that is how I came across this `and/or` comment.

Reply:
Want me to create a separate issue and branch for this or just go ahead and add these 3 checks?